src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoderEncoderViewModel.cs

```diff
@@ -0,0 +1,653 @@
+                Header = JsonHelper.Format(@"{""alg"": """ + algorithm.DisplayName + @""", ""typ"": ""JWT""}", Indentation.TwoSpaces, false);
+                IsSignatureRequired(algorithm);
+                return _algorithmSelected ?? JwtAlgorithmDisplayPair.HS256;
+            }
+            set
+            {
+                if (_algorithmSelected != value)
+                {
+                    SettingsProvider.SetSetting(JwtDecoderEncoderSettings.JwtAlgorithm, value.Value);
+                    IsSignatureRequired(value);
+                    SetProperty(ref _algorithmSelected, value);
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        internal IReadOnlyList<JwtAlgorithmDisplayPair> Algorithms = new ObservableCollection<JwtAlgorithmDisplayPair> {
+            JwtAlgorithmDisplayPair.HS256, JwtAlgorithmDisplayPair.HS384, JwtAlgorithmDisplayPair.HS512,
+            JwtAlgorithmDisplayPair.RS256, JwtAlgorithmDisplayPair.RS384, JwtAlgorithmDisplayPair.RS512,
+            JwtAlgorithmDisplayPair.ES256, JwtAlgorithmDisplayPair.ES384, JwtAlgorithmDisplayPair.ES512,
+            JwtAlgorithmDisplayPair.PS256, JwtAlgorithmDisplayPair.PS384, JwtAlgorithmDisplayPair.PS512,
+        };
+
+        public JwtDecoderEncoderViewModel(
+            ISettingsProvider settingsProvider,
+            IMarketingService marketingService)
+        {
+            SettingsProvider = settingsProvider;
+            MarketingService = marketingService;
+            InputFocusChanged = ControlFocusChanged;
+            IsSignatureRequired(AlgorithmMode);
+        }
+
+        internal void QueueNewTokenJob()
+        {
+            JwtValidation = new ValidationBase
+            {
+                IsValid = true
+            };
+            var newJob = new JwtJobAddedMessage();
+            Messenger.Send(newJob);
+        }
+
+        protected void DisplayValidationInfoBar()
+        {
+            InfoBarSeverity infoBarSeverity;
+            string message;
+            if (!JwtValidation.IsValid)
+            {
+                infoBarSeverity = InfoBarSeverity.Error;
+                message = JwtValidation.ErrorMessage ?? LocalizedStrings.JwtInValidMessage;
+            }
+            else
+            {
+                infoBarSeverity = InfoBarSeverity.Success;
+                message = LocalizedStrings.JwtIsValidMessage;
+            }
+
+            ValidationResult = new InfoBarData(infoBarSeverity, message);
+        }
+
+        protected void ControlFocusChanged(object source, RoutedEventArgs args)
+        {
+            var input = (CustomTextBox)source;
+
+            if (input.Text.Length == 0)
+            {
+                return;
+            }
+
+            QueueNewTokenJob();
+        }
+
+        #region Decoding
+
+        protected SigningCredentials GetValidationCredentials(JwtAlgorithm algorithmMode)
```

Comment:
Looks like this method (and therefore `GetHmacShaValidationKey` and a few other below) isn't used. Should we remove it?